### PR TITLE
Nuke: adding nukeassist 

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Brief description
Adding support for NukeAssist

## Description
For support of NukeAssist we had to limit some Nuke features since NukeAssist itself Nuke with limitations. We do not support Creator and Publisher. User can only Load versions with version control. User can also set Framerange and Colorspace.

## Testing notes:
1. Set NukeAssist to your project
2. Open NukeAssist on task and see the OpenPype menu items have are redefined.

### CODE REVIEW

File: openpype/hosts/nuke/api/pipeline.py [hunk 1]
```diff
@@ -218,7 +219,9 @@ def _install_menu():
     )
     Context.context_label = label
     context_action = menu.addCommand(label)
-    context_action.setEnabled(False)
+
+    if not ASSIST:
```
Comment (iLLiCiTiT) [inline openpype/hosts/nuke/api/pipeline.py:223]: The context action probably should not be added at all when in assist.
```
if not ASSIST:
    context_action = menu.addCommand(label)
    context_action.setEnabled(False)
    menu.addSeparator()
```

File: openpype/hosts/nuke/hooks/pre_nukeassist_setup.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from openpype.lib import PreLaunchHook
+
+class PrelaunchNukeAssistHook(PreLaunchHook):
+    """
+    Adding flag when nukeassist
+    """
+    app_groups = ["nukeassist"]
+
+    def execute(self):
+        self.launch_context.env["NUKEASSIST"] = "1"
```
Comment (iLLiCiTiT) [inline openpype/hosts/nuke/hooks/pre_nukeassist_setup.py:10]:
```suggestion
        self.launch_context.env["NUKEASSIST"] = "1"

```

File: openpype/hosts/nuke/hooks/pre_nukeassist_setup.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from openpype.lib import PreLaunchHook
+
+class PrelaunchNukeAssistHook(PreLaunchHook):
```
Comment (iLLiCiTiT) [inline openpype/hosts/nuke/hooks/pre_nukeassist_setup.py:3]:
```suggestion


class PrelaunchNukeAssistHook(PreLaunchHook):
```

Comment (jakubjezek001) [reply openpype/hosts/nuke/hooks/pre_nukeassist_setup.py:3]: what is the difference? I am perhaps blind, but it seems to be the same.

File: openpype/hosts/nuke/api/pipeline.py (hunk 1 above)
Comment (jakubjezek001) [reply openpype/hosts/nuke/api/pipeline.py:223]: I believe only issue is that we are not able to disable the menu item. Not to create one.

**Pull request merged**